scripts/ci/test_reborn_pr_test_plan.py

```
@@ -635,6 +635,28 @@ def test_sandbox_docker_paths_preserve_regular_test_inventory_selection(
+                plan = self.plan("pull_request", [path])
+                self.assertEqual(plan["mode"], "selected")
+                self.assertTrue(plan["run_sandbox_docker"])
+                self.assertEqual(plan["crate_buckets"], [])
+                self.assertEqual(plan["root_partitions"], [])
+                self.assertEqual(plan["integration_lanes"], [])
+
+        unrelated_path = "docker/unrelated-helper"
+        with self.subTest(path=unrelated_path):
+            with self.assertRaisesRegex(
+                ValueError, "unclassified pull-request path"
+            ):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)